### DISCUSSION TITLE
how to animate an icon layer

### DISCUSSION THREAD

**Discussion by bjvesely** [visgl/deck.gl#5796] [Q&A]
Can someone please give me any pointers on how to animate an icon layer?  I am trying to cycle through icons every few seconds so that it appears there is animation going on.  Here is sample of layering code below.

new IconLayer({
        id: this.id 
        data: this.props.data,
        sizeUnits: "meters",
        iconAtlas: this.props.iconAtlas,
        iconMapping: this.props.iconMapping,
        pickable: true,
        autoHighlight: true,
        getIcon: (d) => //change icon at every second
}

Thank you,
Brad

**Comment by padawannn** [comment]
Take a look at the [update triggers](https://deck.gl/docs/developer-guide/performance#use-updatetriggers)
